### ISSUE TITLE
ListenBrainz plugin

### ISSUE THREAD

**Issue by sampsyo** [beetbox/beets#1719]
We should add a plugin to integrate with [ListenBrainz](https://listenbrainz.org). ListenBrainz is a new MetaBrainz project similar to Last.fm: it tracks your listening habits.

An initial version could look very much like the existing [lastimport](http://docs.beets.io/en/v1.3.15/plugins/lastimport.html) plugin: just download listen counts and such for music in your database. The data would be stored in flexible attributes.

**Comment by bearcatsandor** [comment]
This would be a fantastic addition. Thank you.

**Comment by pprkut** [comment]
I think this would be well suited as a metasync source rather than a separate new plugin.

**Comment by sampsyo** [comment]
Hmm… good point, @pprkut. The current metasync backends are for local players, but I don't see any immediate reason why a service like ListenBrainz wouldn't fit the same mold.

**Comment by xsteadfastx** [comment]
Oh. This sounds like a nice project. I wonder how it differs from [libre.fm](https://libre.fm/). Last.fm feels dead somehow. I maybe will have a look on it and the lastimport plugin.

**Comment by claman** [comment]
I'm definitely interested to see how this progresses

**Comment by claman** [comment]
I also like this idea

**Comment by justinmayer** [comment]
Now that the [ListenBrainz beta has been launched](https://blog.musicbrainz.org/2017/07/28/listenbrainz-enters-beta-stage/), anyone want to volunteer to work on a ListenBrainz plugin for Beets? 😉

**Comment by xsteadfastx** [comment]
i really down get the api docs on listenbrainz. where to send what?

**Comment by the-solipsist** [comment]
In case someone is interested in developing a plugin, here's the format for the Listenbrainz [JSON payload](https://listenbrainz.readthedocs.io/en/latest/dev/json/), and here's the [API](https://listenbrainz.readthedocs.io/en/latest/dev/api/).

**Comment by Kernald** [comment]
Did anyone ever made any progress on that? I don't have any Python experience myself but might give that a shot. There's now a Python API library ([`pylistenbrainz`](https://pypi.org/project/pylistenbrainz/)), which should make this fairly straightforward, starting from `lastimport` or `metasync`.

**Comment by sampsyo** [comment]
Sounds awesome!

**Comment by arsaboo** [comment]
@sampsyo I created a basic [Listenbrainz plugin](https://github.com/arsaboo/beets-listenbrainz). My primary purpose was to import the weekly playlist recommendations into Plex. I did not use the pylistenbrainz library for now as 1) it does not have the playlist features that I am looking for, and 2) the basic API is very simple to be implemented directly. 

Let me know if this looks fine and I will be happy to submit it to the core so that we can build upon this.
